indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskRunner.java

```diff
@@ -124,14 +125,14 @@ default TaskLocation getTaskLocation(String taskId)
 
   /**
    * APIs useful for emitting statistics for @TaskSlotCountStatsMonitor
-  */
-  long getTotalTaskSlotCount();
+   */
+  Map<String, Long> getTotalTaskSlotCount();
```

Comment:
nit: could use `Object2LongMap`.

Reply:
Since I'm returning `ImmutableMap` from methods where there will be only one entry, I can't use `Object2LongMap` without changing return values.

Reply:
I'm not sure what you mean. My suggestion is changing all return type here including all their implementations. For singleton map, you can use `Object2LongMaps.singleton()`. But this comment is nit because this api is not performance critical.

Reply:
If it's okay I'm not planning to address this change in this PR.